src/components/stat-plot/NewStatPlot.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/jsx-filename-extension */
+import React from 'react';
+import styled from 'styled-components';
+import colors from '../../constants/colors';
+import fontSizes from '../../constants/fontSizes';
+import fontWeights from '../../constants/fontWeights';
+import radii from '../../constants/radii';
+import space from '../../constants/space';
+import {
+  getAxisName,
+  getSubStatDatumByDataKey,
+  getDoesSubStatDataExist,
+  getStatDataKeys,
+  getStatDatumByDataKey,
+  getSubStatAxis,
+  getTickFormatterFromDataKey,
+} from '../../utils/plotConfigGetters';
+
+function NewStatPlot({ plotConfig = {} }) {
+  const statDataKeys = getStatDataKeys(plotConfig);
+  const doesSubStatDataExist = getDoesSubStatDataExist(plotConfig);
+  const numMetrics = statDataKeys.length;
+  const showBorder = numMetrics !== 1;
+
+  return (
+    <StatsList numMetrics={numMetrics}>
+      {statDataKeys.map((statDataKey) => {
+        const tickFormatter = getTickFormatterFromDataKey(plotConfig, statDataKey);
+        const datum = getStatDatumByDataKey(plotConfig, statDataKey);
+        const value = datum?.[statDataKey];
+        const axisName = getAxisName(plotConfig, statDataKey);
+
+        return (
+          <Stat showBorder={showBorder} key={statDataKey}>
+            <Label>{axisName}</Label>
+            <Value>{tickFormatter(value) ?? '-'}</Value>
+            {doesSubStatDataExist && <SubStat statDataKey={statDataKey} plotConfig={plotConfig} />}
+          </Stat>
+        );
+      })}
+    </StatsList>
+  );
+}
+
+function SubStat({ plotConfig, statDataKey }) {
+  const subStatAxis = getSubStatAxis(plotConfig);
+  const subStatData = getSubStatDatumByDataKey(plotConfig, statDataKey);
+  return subStatAxis.format(subStatData);
+}
+
+const getStatGridCss = (numMetrics) => {
```

Comment:
Before I was doing some intense calculation to handle the # of columns **and** the column gap, but I think it's clearer instead to just fork on whether we do `grid-auto-flow` or `grid-template-columns`, which take the column gap into account.